index.js

```diff
@@ -19,10 +28,76 @@ module.exports = {
         return converter.convert(data, options, cb);
       });
     }
-    return cb(null, {
-      result: false,
-      reason: 'input type:' + input.type + ' is not valid'
-    });
+    else if (input.type === 'folder') {
+      let filesPathArray = input.data,
+        convertedSpecs = [],
+        rootFiles = parse.getRootFiles(filesPathArray);
+
+      async.eachSeries(rootFiles, (rootFile, callback) => {
+        parse
+        // will merge all the files in the folder
+          .mergeFiles(rootFile, OasResolverOptions)
+          .then((spec) => {
+            converter.convert(spec, options, (err, result) => {
+              if (err) {
+                return callback(null, {
+                  result: false,
+                  reason: err
+                });
+              }
+              // eslint-disable-next-line no-else-return
+              else {
+                convertedSpecs.push(result);
+                return callback(null);
+              }
+
+            });
+          })
+          .catch((err) => {
+            return callback(null, {
+              result: false,
+              reason: err
+            });
+          });
+      }, () => {
+
+        var conversionResult = false,
+          convertedCollections = [],
+          reasonForFail;
+
+        _.forEach(convertedSpecs, (convertedSpec) => {
+          if (convertedSpec.result) {
+            conversionResult = convertedSpec.result;
+            convertedCollections.push(convertedSpec.output[0]);
+          }
+          else {
+            conversionResult = convertedSpec.result;
+            reasonForFail = convertedSpec.reason;
```

Comment:
you're overriding the variable in each instance of the loop. The value of reasonForFail in line 89 will just be the reason for the last failure

Reply:
Same for conversionResult. Why does it need to be overriden in each iteration of the loop?

Reply:
Yes..I've added the break statement if any of the `convertedResult.result` is false